hydra-node/src/Hydra/HeadLogic.hs

```diff
@@ -367,9 +385,12 @@ data LogicError tx
   = InvalidEvent (Event tx) (HeadState tx)
   | InvalidState (HeadState tx)
   | LedgerError ValidationError
+  deriving stock (Generic)
 
 instance Tx tx => Exception (LogicError tx)
 
+deriving instance Tx tx => ToJSON (LogicError tx)
+deriving instance Tx tx => FromJSON (LogicError tx)
```

Comment:
this file is getting huge, we really need to think about separating the `Types` from the logic. Putting something in the red bin...

Reply:
I agree. Many things in there are not really about the head logic. I'd rather separate things into modules for each type (or at least, logical groups of types) and not into a gigantic `Types.hs` module if we can avoid it :blush: